handlers/anteHandler.go

```diff
@@ -0,0 +1,183 @@
+package handlers
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/FourthState/plasma-mvp-sidechain/msgs"
+	"github.com/FourthState/plasma-mvp-sidechain/plasma"
+	"github.com/FourthState/plasma-mvp-sidechain/store"
+	"github.com/FourthState/plasma-mvp-sidechain/utils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"math/big"
+)
+
+// FeeUpdater updates the aggregate fee amount in a block
+type FeeUpdater func(amt *big.Int) sdk.Error
+
+type plasmaConn interface {
+	GetDeposit(*big.Int) (plasma.Deposit, bool)
+	HasTxBeenExited(plasma.Position) bool
+}
+
+func NewAnteHandler(utxoStore store.UTXOStore, plasmaStore store.PlasmaStore, feeUpdater FeeUpdater, client plasmaConn) sdk.AnteHandler {
+	return func(ctx sdk.Context, tx sdk.Tx, simulate bool) (newCtx sdk.Context, res sdk.Result, abort bool) {
+		spendMsg, ok := tx.(msgs.SpendMsg)
+		if !ok {
+			return ctx, sdk.ErrInternal("tx must in the form of a spendMsg").Result(), true
+		}
+
+		txHash := spendMsg.TxHash()
+		var totalInputAmt, totalOutputAmt *big.Int
+
+		/* validate the first input */
+		amt, res := validateInput(ctx, spendMsg.Input0, true, spendMsg.Fee, utxoStore, client)
+		if !res.IsOK() {
+			return ctx, res, true
+		}
+		res = validateSignatures(ctx, spendMsg.Input0, txHash, utxoStore)
+		if !res.IsOK() {
+			return ctx, res, true
+		}
+		if client.HasTxBeenExited(spendMsg.Input0.Position) {
```

Comment:
Shouldn't we also check that inputs parents haven't exited?

Reply:
You are right. Listing the reason here to document later

For the case. 
1.) I have an output A
2.) I broadcast a spend to myself A->B
3.) Exit A (no one can challenge. I keep the confirm signature)
4.) Wait for the exit to finish
5.) Spend B (should not be allowed ever)
6.) B cannot be exited due an exited input
7.) B becomes a stale utxo which is okay

Reply:
Was this ever updated? I don't see the parent inputs being checked?

Reply:
The parents are checked in `validateInput` line 155.